Tests/Components Tests/Gift Card/GiftCardComponentTests.swift

```diff
@@ -46,12 +46,12 @@ class GiftCardComponentTests: XCTestCase {
 
     override func setUpWithError() throws {
         try super.setUpWithError()
-        paymentMethod = GiftCardPaymentMethod(type: .giftcard, name: "testName", brand: "testBrand")
+        giftCardPaymentMethod = GiftCardPaymentMethod(type: .giftcard, name: "testName", brand: "testBrand")
         publicKeyProvider = PublicKeyProviderMock()
 
         context = Dummy.context
 
-        sut = GiftCardComponent(paymentMethod: paymentMethod,
+        sut = GiftCardComponent(partialPaymentMethodType: .giftCard(giftCardPaymentMethod),
```

Comment:
Maybe good to stick with the public convenience init here to have that flow covered

```suggestion
        sut = GiftCardComponent(paymentMethod: giftCardPaymentMethod,
```

Reply:
But then the public key provider won't be a mock. So we need this one here but we do use other init paths in other tests anyway.